### PR TITLE
feat: button component

### PR DESCRIPTION
**Pull request by sugarpusher**
Updated button component

### CODE REVIEW

**Review by vojtechsimetka — COMMENTED** [review]
Hi @sugarpusher great job! In general the code is fine IMHO (probably @filoozom will be more opinionated :) ) but I would like to ask for few improvements.

1. I don't think it makes sense to change the `color` and `bgColor` independently, right? So I would suggest that we drop the `bgColor` and the `color` attribute also applies adequate backgroud color.
2. In the disabled state, the cursor still indicates that the button can be interacted with. Would be great to fix that.
3. This is more for @filoozom - IMHO we should pass to the `class` argument not only the `className` but also the `class`. So in `src/button/button.tsx` we should write `class: [class, className].join(' ')`,

I'll let @filoozom to approve this one but will be happy to be your review buddy for future PRs :)

**Comment by sugarpusher** [comment]
Thanks for the feedback @vojtechsimetka, that's perfect!

1. I left the text and background colors separate mostly because we have text color variations over white background, sometimes the text is black, sometimes it can be red. I don't remember if there are any other instances right now but that's mainly why I made them separate. Maybe it makes sense then to create more specific variations for those instances then? 
2. I missed that, @filoozom had set up the disable state and I didn't even touch it. I'll check it.
3. That wasn't for me 😁

Thanks again!

**Comment by filoozom** [comment]
The code is good and I like that you added the custom Storybook buttons. To answer the comments above:

1. I tend to agree with @vojtechsimetka. Giving users of the library the choice of using bad `color`/`bgColor` combinations might not be great. But in the great scheme of things I'd say it doesn't really matter and is a tiny detail.
2. That is indeed my bad
3. Sure, but does it really matter? In Preact we'd use `class` all the time anyways. Do we need to be compatible with React? I don't care too much either way.

Before we merge, there does seem to be an issue with formatting. but I'm just now realizing that I haven't set any linting / prettier config up. Maybe I should quickly add those so we can reformat all the code in this PR, or I can also do it afterwards, it doesn't really matter.

One (tiny) nit-pick: I'm not too sure why `settings.json` was added to `.gitignore`, but in general I'd say this should have been done in a separate PR.

Let's decide on what we'll do regarding point 1, and I'll approve once we have consensus 🙂

File: src/button/button.module.css
Comment (sugarpusher) [reply src/button/button.module.css:12]: In the design, the button is never smaller than that, regardless of the text inside. It's actually almost always that exact size, except in rare cases when the text is longer. We also don't want the text to wrap so I didn't want to use a fixed width.

**Review by filoozom — APPROVED** [review]
Looks good to me! Let's wait for #2 to be merged so we can refactor this PR and apply the same styling before merging though. 🙂

**Pull request merged**
